sdk/cpp/core/src/gnmi_provider.cpp

```diff
@@ -21,273 +21,271 @@
+    static shared_ptr<path::Rpc> create_rpc_instance(path::RootSchemaNode & root_schema, string rpc_name);
+    static path::DataNode& create_rpc_input(path::Rpc & gnmi_rpc);
+    static void create_input_source(path::DataNode & input, bool config);
+    static string get_read_rpc_name(bool config);
+    static bool is_config(path::Rpc & rpc);
+    static string get_filter_payload(path::Rpc & ydk_rpc);
+    static string get_gnmi_payload(path::DataNode & input, string data_tag, string data_value);
+    static std::shared_ptr<path::DataNode> handle_read_reply(string reply, path::RootSchemaNode & root_schema);
+    static gNMIServiceProvider::SecureChannelArguments get_channel_credentials();
+
+    // Create a default SSL ChannelCredentials object.
+    gNMIServiceProvider::SecureChannelArguments input_args = get_channel_credentials(); 
+
+    gNMIServiceProvider::gNMIServiceProvider(string address)
+        : client(make_unique<gNMIClient>(grpc::CreateCustomChannel(address, input_args.channel_creds, input_args.args)))
+    {
+        path::Repository repo;       
+        initialize(repo, address);
+        YLOG_INFO("Connected to {} using ssh", address);
+    }
+
+    gNMIServiceProvider::gNMIServiceProvider(path::Repository & repo, string address)
+        : client(make_unique<gNMIClient>(grpc::CreateCustomChannel(address, input_args.channel_creds, input_args.args)))
+    {
+        initialize(repo, address);
+        YLOG_INFO("Connected to {} using ssh", address);
+    }
+
+    gNMIServiceProvider::~gNMIServiceProvider() = default;
+
+    void gNMIServiceProvider::initialize(path::Repository & repo, std::string address) {
+        IetfCapabilitiesParser capabilities_parser{};
+        client->connect(address);
+        server_capabilities = client->get_capabilities();
+        
+        std::cout << "In gNMIServiceProvider initialize" << std::endl;
+        root_schema = repo.create_root_schema(capabilities_parser.parse(server_capabilities));
+
+        if(root_schema.get() == nullptr)
+        {
+            YLOG_ERROR("Root schema cannot be obtained");
+            throw(YCPPIllegalStateError{"Root schema cannot be obtained"});
+        }
+    }
+
+    gNMIServiceProvider::SecureChannelArguments get_channel_credentials() 
+    {
+        std::string server_cert, client_key, client_cert;
+        std::ifstream rf("ems.pem");
+        //std::ifstream kf("client.key");
+        //std::ifstream cf("client.pem");
+
+        server_cert.assign((std::istreambuf_iterator<char>(rf)),(std::istreambuf_iterator<char>()));
+        //client_key.assign((std::istreambuf_iterator<char>(kf)),(std::istreambuf_iterator<char>()));
+        //client_cert.assign((std::istreambuf_iterator<char>(cf)),(std::istreambuf_iterator<char>()));
+
+        grpc::SslCredentialsOptions ssl_opts;
+        grpc::ChannelArguments      args;
+        gNMIServiceProvider::SecureChannelArguments input_args;
+
+        std::cout << "server cert: " << server_cert << std::endl;
```

Comment:
True, removed this in [66c08bb](https://github.com/shgandhi/ydk-gen/commit/66c08bbe5f1873c0d1dac73d75f813a84c587cf8)